pkg/agent/route/route_windows.go

```diff
@@ -177,6 +205,105 @@ func (c *Client) DeleteRoutes(podCIDR *net.IPNet) error {
 	return nil
 }
 
+// addServiceVirtualIPToGWRoutes adds routes on a Windows Node for redirecting ClusterIP Service traffic
+// from host network to OVS via antrea-gw0.
+func (c *Client) addServiceVirtualIPToGWRoutes() {
+	metric := util.MetricHigh
+	linkIndex := c.nodeConfig.GatewayConfig.LinkIndex
+	gwIP := c.nodeConfig.GatewayConfig.IPv4
+
+	// A virtual IP is used to avoid too many neighbor cache entries.
+	// Route: Virtual IP (169.254.169.252) -> antrea-gw0
+	vRoute := &util.Route{
+		LinkIndex:         linkIndex,
+		DestinationSubnet: &serviceGWVirtualIPNet,
+		GatewayAddress:    gwIP,
+		RouteMetric:       metric,
+	}
+	util.NewNetRoute(vRoute)
+
+	// For ClusterIP Service requests from host, reply traffic needs a route entry to route packet to ServiceGWVirtualIP via antrea-gw0.
+	// If the NextHop of a route is antrea-gw0, then set it as 0.0.0.0. As a result, on-link/0.0.0.0 is in PersistentStore.
+
+	repRoute := &util.Route{
+		LinkIndex:         linkIndex,
+		DestinationSubnet: &config.ServiceGWHairpinIPv4Net,
+		GatewayAddress:    net.IPv4zero,
+		RouteMetric:       metric,
+	}
+	util.NewNetRoute(repRoute)
+}
+
+func (c *Client) AddClusterIPRoute(svcIP net.IP) error {
```

Comment:
I think you could also add ClusterIP route like the way in Linux - just maintain one ClusterIP route. 
- If there is a route for ClusterIP, add one
- If current route can route the new added ClusterIP, do nothing
- If current route cannot route the new added ClusterIP, add a new route and remove the old one. 
    - Calculate a new IP block with current route destination and the ClusterIP. 
    - Use the new calculated IP block as destination for a new route, which is able to route the ClusterIP.

Reply:
It is a good idea and I will update eventually, maybe in this PR or in a new one. I don't want to block the review process.